File: TypeScript/UmActually/umActually.ts
const umActually = (phrase: string): string => {
  const loweredPhrase: string = phrase.toLowerCase();


  let finalPhrase: string = '';

  for (let i = 0; i < loweredPhrase.length; i++) {
    if (loweredPhrase[i].match(/[^A-Za-z0-9]/g)) {
      finalPhrase += loweredPhrase[i];
    } else {
      if (i % 2 == 0) {
        finalPhrase += loweredPhrase[i].toUpperCase()
      } else {
        finalPhrase += loweredPhrase[i];
      }
    }
  }

  console.log(finalPhrase);
  return phrase;
}

umActually('Inertia is a property of matter')

File: Java/UmActually/UmActually.java
package Java.UmActually;

public class UmActually {
  public static String UmActually(String phrase) {
    String loweredPhrase = phrase.toLowerCase();
    String finalPhrase = "";

    for (int i = 0; i < loweredPhrase.length(); i++) {
        if (Character.isLetter(loweredPhrase.charAt(i))) {
            if (i % 3 == 0) {
                finalPhrase += Character.toUpperCase(loweredPhrase.charAt(i));
            } else {
                finalPhrase += loweredPhrase.charAt(i);
            }
        } else {
            finalPhrase += loweredPhrase.charAt(i);
        }
    }

    return finalPhrase;
}
  public static void main(String[] args) {
    System.out.println(UmActually("Inertia is a property of matter"));
  }
}
